packages/web-opds-client/src/components/__tests__/OPDSCatalog-test.tsx

```diff
@@ -48,4 +49,25 @@ describe("OPDSCatalog", () => {
       expect(root.props()[key]).to.equal(props[key]);
     });
   });
+
+  it("passes custom DataFetcher to Root", () => {
+    const customFetcher = new DataFetcher();
+    const propsWithFetcher = {
+      ...props,
+      fetcher: customFetcher
+    };
+
+    let wrapper = shallow(<OPDSCatalog {...propsWithFetcher} />, {
+      context,
+      childContextTypes: {
+        router: PropTypes.object,
+        pathFor: PropTypes.func
+      }
+    });
+
+    const root = wrapper.dive().dive().dive().dive();
```

Comment:
The use of multiple `dive`s here feels really brittle. Any reason not to just fully `mount` (rather than `shallow` mount) the `OPDSCatalog` component and then find `Root` in the DOM before checking the props?

```suggestion
    const wrapper = mount(<OPDSCatalog {...propsWithFetcher} />, {
      context,
      childContextTypes: {
        router: PropTypes.object,
        pathFor: PropTypes.func
      }
    });

    const root = wrapper.find(Root);

    // Verify that we found exactly one `Root` component.
    expect(root).to.have.length(1);
```

Reply:
Thanks Tim. Good suggestion. This was just my unfamiliarity with the test framework.